02-microarray/pathway-analysis_microarray_03_gsva.Rmd

```diff
@@ -364,48 +367,42 @@ max_dup_df <- mapped_df %>%
 
 max_dup_df
 ```
-Looks like we were able to successfully get rid of the duplicate gene identifiers and keep the observations with the maximum expression value!
+Looks like we were able to successfully collapse the duplicate gene identifiers by keeping the maximum expression value per sample associated with each of the duplicate gene identifiers!
 
 Now let's combine the rest of the mapped data!
 
 ```{r}
 filtered_mapped_df <- mapped_df %>%
-  dplyr::filter(
-    # First let's get the data associated with the gene identifiers that aren't duplicated
-    !entrez_id %in% dup_entrez_ids |
-      # Now let's get the data associated with the duplicated gene identifiers that we
-      # collapsed in `max_dup_df`
-      entrez_id %in% max_dup_df$entrez_id &
-        Ensembl %in% max_dup_df$Ensembl
-  )
+  # First let's get the data associated with the gene identifiers that aren't duplicated
+  dplyr::filter(!entrez_id %in% dup_entrez_ids) %>%
+  # Now let's bind the rows of the data we collapsed in `max_dup_df`
+  dplyr::bind_rows(max_dup_df) %>%
+  # Arrange so that the Entrez IDs that were duplicated are at the top
+  dplyr::arrange(dplyr::desc(entrez_id %in% dup_entrez_ids)) %>%
```

Comment:
Is there a reason we want the duplicates on the top? I think mixing in is fine? Or leaving as is?

Reply:
The reason behind this was in preparation of the next step, where we can use the `head()` function to show that we retained the collapsed duplicate data as well as the rest of the mapped data.